src/main/java/org/zalando/nakadi/service/subscription/IdleStreamWatcher.java

```diff
@@ -0,0 +1,36 @@
+package org.zalando.nakadi.service.subscription;
+
+public class IdleStreamWatcher {
```

Comment:
it is not clear why this class is synchronised at all, it is used in StreamingState, which runs in one thread.

Reply:
I think it doesn't always run in one Thread. It has code which is executed from ZK-Watcher threads.